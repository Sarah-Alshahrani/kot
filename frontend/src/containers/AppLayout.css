html,
body {
    height: 60%;
}