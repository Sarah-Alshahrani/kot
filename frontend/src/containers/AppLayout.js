import React, { Component } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import GameboardLayout from "./Gameboard";
import LobbyLayout from "./Lobby";
import LoginLayout from "./Login";
import NotFoundPage from "./../components/NotFoundPage/NotFoundPage";

import "./AppLayout.css";

class AppLayout extends Component {
  state = {
    collapsed: false
  };

  onCollapse = collapsed => {
    this.setState({ collapsed });
  };

  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed
    });
  };

  render() {
    return (
      <Router>
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
          <a className="navbar-brand" href="/login">
            Home
          </a>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarNavDropdown"
            aria-controls="navbarNavDropdown"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarNavDropdown">
            <ul className="navbar-nav">
              <li className="nav-item">
                <a className="nav-link" href="/login">
                  Login
                </a>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="/lobby">
                  Lobby <span className="sr-only">(current)</span>
                </a>
              </li>
              <li className="nav-item">
                <a className="nav-link" href="/game">
                  Play KoT
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <Switch>
          <Route exact path="/" component={LobbyLayout} />
          <Route path="/login" component={LoginLayout} />
          <Route path="/game" component={GameboardLayout} />
          <Route path="/lobby" component={LobbyLayout} />
          <Route component={NotFoundPage} />
        </Switch>
      </Router>
    );
  }
}

export default AppLayout;
