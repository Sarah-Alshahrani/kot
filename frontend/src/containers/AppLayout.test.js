import React from 'react';
import { shallow } from 'enzyme';

import AppLayout from './AppLayout';

// verify lobby and play ToK
import {
    BrowserRouter as Router,
    Route,
    Link,
    Switch
} from 'react-router-dom';

describe('Verify site layout template', () => {
    it('General layout should render correctly', () => {
        const errorPage = shallow(<AppLayout />);
        expect(errorPage).toMatchSnapshot();
    });

    it('renders lobby menu', () => {
        const wrapper = shallow(<AppLayout />);

        const login_menu = <li className="nav-item">
            <a className="nav-link" href="/login">Login</a>
        </li>

        const lobby_menu = <li className="nav-item">
            <a className="nav-link" href="/lobby">Lobby <span className="sr-only">(current)</span></a>
        </li>

        const game_menu = <li className="nav-item">
            <a className="nav-link" href="/game">Play KoT</a>
        </li>

        expect(wrapper).toContainReact(login_menu);
        expect(wrapper).toContainReact(lobby_menu);
        expect(wrapper).toContainReact(game_menu);
    });
})