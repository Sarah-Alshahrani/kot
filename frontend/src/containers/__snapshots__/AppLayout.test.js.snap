// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify site layout template General layout should render correctly 1`] = `
<BrowserRouter>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/login"
    >
      Home
    </a>
    <button
      aria-controls="navbarNavDropdown"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNavDropdown"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavDropdown"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/lobby"
          >
            Lobby 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/game"
          >
            Play KoT
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/game"
    />
    <Route
      component={[Function]}
      path="/lobby"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
